File: app/src/main/java/com/example/worldvisit/Validation.java
package com.example.worldvisit;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.CalendarView;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.Calendar;

public class Validation extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_validation);

        final CalendarView calendarView = (CalendarView) findViewById(R.id.calendar); // get the reference of CalendarView
        TextView day = (TextView) findViewById(R.id.dayVisit);
        final Calendar c = Calendar.getInstance();
        int year = c.get(Calendar.YEAR);
        int month = c.get(Calendar.MONTH);
        int dayOfMonth = c.get(Calendar.DAY_OF_MONTH);
        //on récupère le numéro du jour dans la semaine
        int wDay=c.get(Calendar.DAY_OF_WEEK);
        String dayOfWeek = new String();
        String Month = new String();
        switch (wDay){
            case 0:
                dayOfWeek = "Dim";
            case 1:
                dayOfWeek = "Lun";
            case 2:
                dayOfWeek = "Mar";
            case 3:
                dayOfWeek = "Mer";
            case 4:
                dayOfWeek = "Jeu";
            case 5:
                dayOfWeek = "Ven";
            case 6:
                dayOfWeek = "Sam";
        }
        switch (month){
            case 0:
                Month = "Janvier";
            case 1:
                Month = "Fevrier";
            case 2:
                Month = "Mars";
            case 3:
                Month = "Avril";
            case 4:
                Month = "Mai";
            case 5:
                Month = "Juin";
            case 6:
                Month = "Juillet";
            case 7:
                Month = "Août";
            case 8:
                Month = "Septembre";
            case 9:
                Month = "Octobre";
            case 10:
                Month = "Novembre";
            case 11:
                Month = "Décembre";
        }

        // Traitement de l'affichage du jours et du mois (qui ne marche pas ...)
        String displayedDay = dayOfWeek+". "+dayOfMonth+" "+Month+". "+year;
        day.setText(displayedDay);
        // perform setOnDateChangeListener event on CalendarView
        calendarView.setOnDateChangeListener(new CalendarView.OnDateChangeListener() {

            @Override
            public void onSelectedDayChange(CalendarView view, int year, int Month, int dayOfMonth) {
                long selectedDate = calendarView.getDate();
            }
        });
    }

    public void backToHome(View view) {
        finish();
        Intent intent = new Intent(this, MainActivity.class);
        /*intent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);*/
        startActivity(intent);
    }
}

File: app/src/main/java/com/example/worldvisit/Validation.kt
package com.example.worldvisit

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.CalendarView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.worldvisit.database.DatabaseHelper
import java.util.*

class Validation : AppCompatActivity() {
    private var databaseHelper: DatabaseHelper? = null
    var date = String()

    override fun onCreate(savedInstanceState: Bundle?) {
        databaseHelper = DatabaseHelper(this)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_validation)
        val calendarView = findViewById<View>(R.id.calendar) as CalendarView
        val day = findViewById<View>(R.id.dayVisit) as TextView
        val c = Calendar.getInstance()
        val year = c[Calendar.YEAR]
        val monthofYear = c[Calendar.MONTH]
        val dayOfMonth = c[Calendar.DAY_OF_MONTH]
        val wDay = c[Calendar.DAY_OF_WEEK]
        var dayOfWeek = String()
        var month = String()
        Log.i("TAG", "${Calendar.DAY_OF_WEEK}")
         when (wDay){
            0 -> dayOfWeek = "Sam";
            1 -> dayOfWeek = "Dim";
            2 -> dayOfWeek = "Lun";
            3 -> dayOfWeek = "Mar";
            4 -> dayOfWeek = "Mer";
            5 -> dayOfWeek = "Jeu";
            6 -> dayOfWeek = "Ven";
        }
        when (monthofYear){
            0 -> month = "Janvier";
            1 -> month = "Fevrier";
            2 -> month = "Mars";
            3 -> month = "Avril";
            4 -> month = "Mai";
            5 -> month = "Juin";
            6 -> month = "Juillet";
            7 -> month = "Août";
            8 -> month = "Septembre";
            9 -> month = "Octobre";
            10 -> month = "Novembre";
            11 -> month = "Décembre";
        }

        val displayedDay = "$dayOfWeek. $dayOfMonth $month $year"
        date = displayedDay.format("$dayOfMonth ${month[2]} $year")
        //yearView.setText(displayYear);
        day.text = displayedDay
        // perform setOnDateChangeListener event on CalendarView
        calendarView.setOnDateChangeListener { view, year, Month, dayOfMonth ->
            val selectedDate = calendarView.date
        }
    }

    fun backToHome(view: View?) {
        this.databaseHelper!!.addpaysDetail("Espagne", "Madrid", "Europe", "$date", "ES");
        finish()
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
    }
<<<<<<< HEAD
}
=======
}
>>>>>>> origin/master
